common/src/main/java/com/microsoft/identity/common/internal/cache/CacheKeyValueDelegate.java

```diff
@@ -183,6 +185,10 @@ public String generateCacheKey(Credential credential) {
             final IdTokenRecord idToken = (IdTokenRecord) credential;
             cacheKey = cacheKey.replace(REALM, sanitizeNull(idToken.getRealm()));
             cacheKey = cacheKey.replace(TARGET, "");
+        } else if (credential instanceof PrimaryRefreshTokenRecord) {
+            final PrimaryRefreshTokenRecord primaryRefreshTokenRecord = (PrimaryRefreshTokenRecord) credential;
+            cacheKey = cacheKey.replace(REALM, sanitizeNull(primaryRefreshTokenRecord.getRealm()));
```

Comment:
At a glance, this doesn't look correct.... IIRC `RefreshToken` credential types set `""` as the realm, my suspicion is that PRTs should be cached according to the same rules. This should be confirmed with the iOS team for consistency

Reply:
Thanks. Changed REALM to "". And also do we need to set client_id to "" as well for PRT ?

Reply:
@richard1122 please confirm the desired cache-key behavior with @oldalton and write your tests accordingly